tests/store/artifact/test_databricks_artifact_repo.py

```diff
@@ -591,6 +556,226 @@ def test_paginated_list_artifacts(self, databricks_artifact_repo):
+        """
+        credential_infos_mock_1 = [
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_1", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            ),
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_2", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            ),
+        ]
+        credential_infos_mock_2 = [
+            ArtifactCredentialInfo(
+                signed_uri="http://mock_url_3", type=ArtifactCredentialType.AWS_PRESIGNED_URL
+            )
+        ]
+        credential_infos_mock_3 = []
+
+        with mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY_PACKAGE + ".message_to_json"
+        ) as message_mock, mock.patch(
+            DATABRICKS_ARTIFACT_REPOSITORY + "._call_endpoint"
+        ) as call_endpoint_mock:
+            get_credentials_for_read_responses = [
+                GetCredentialsForRead.Response(
+                    credential_infos=credential_infos_mock_1, next_page_token="2"
+                ),
+                GetCredentialsForRead.Response(
+                    credential_infos=credential_infos_mock_2, next_page_token="3"
+                ),
+                GetCredentialsForRead.Response(credential_infos=credential_infos_mock_3,),
+            ]
+            call_endpoint_mock.side_effect = get_credentials_for_read_responses
+            read_credential_infos = databricks_artifact_repo._get_read_credential_infos(
+                MOCK_RUN_ID, ["testpath"],
+            )
+            assert read_credential_infos == credential_infos_mock_1 + credential_infos_mock_2
+            message_mock.assert_has_calls(
+                [
+                    mock.call(GetCredentialsForRead(run_id=MOCK_RUN_ID, path=["testpath"])),
+                    mock.call(
+                        GetCredentialsForRead(run_id=MOCK_RUN_ID, path=["testpath"], page_token="2")
+                    ),
+                    mock.call(
+                        GetCredentialsForRead(run_id=MOCK_RUN_ID, path=["testpath"], page_token="3")
+                    ),
+                ]
+            )
+            assert call_endpoint_mock.call_count == 3
+
+    def test_get_read_credential_infos_respects_max_request_size(self, databricks_artifact_repo):
+        """
+        Verifies that the `_get_read_credential_infos` method, which is used to resolve read access
+        credentials for a collection of artifacts, handles paginated responses properly, issuing
+        incremental requests until all pages have been consumed
+        """
+        # Create 3 chunks of paths, two of which have the maximum request size and one of which
+        # is smaller than the maximum chunk size. Aggregate and pass these to
+        # `_get_read_credential_infos`, validating that this method decomposes the aggregate
+        # list into these expected chunks and makes 3 separate requests
+        paths_chunk_1 = ["path1"] * 2000
```

Comment:
Change 2000 to _MAX_CREDENTIALS_REQUEST_SIZE, ditto below.

Reply:
I'm concerned that this might come back to bite us of `_MAX_CREDENTIALS_REQUEST_SIZE` ends up being erroneously changed to some incorrect value like `3`.  Let me know if you feel strongly about this refactoring though!

Reply:
Hardcoding is fine. It just might break if the size is changed to something bigger later, not a big deal though.

Reply:
Based on offline discussion, I've changed the test to use the conf value and added an assert on the conf value with a descriptive message indicating that it should only be changed in response to a backend limit change